service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftCMOControllerTest.java

```diff
@@ -0,0 +1,148 @@
+@OverrideAutoConfiguration(enabled = true)
+@SuppressWarnings("unchecked")
+class DraftCMOControllerTest {
+
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+    @Autowired
+    private DraftCMOService draftCMOService;
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper mapper;
+
+    private final LocalDate date = LocalDate.now();
+    private final List<Element<HearingBooking>> hearingDetails = createHearingBookings(date);
+
+    @Test
+    void aboutToStartCallbackShouldFillTheHearingDatesList() throws Exception {
+
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .data(ImmutableMap.of("hearingDetails", hearingDetails))
+                .build())
+            .build();
+
+        List<String> expected = Arrays.asList(
+            draftCMOService.convertDate(date.plusDays(5)),
+            draftCMOService.convertDate(date.plusDays(2)),
+            draftCMOService.convertDate(date));
+
+        actAndAssert(request, expected);
+    }
+
+    private void actAndAssert(CallbackRequest request, List<String> expected) throws Exception {
+        MvcResult response = makeRequest(request, "about-to-start");
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = mapper.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        assertThat(getReturnedDatesFromResponse(callbackResponse)).isEqualTo(expected);
+    }
+
+    @Test
+    void aboutToSubmitShouldPopulateHiddenHearingDateField() throws Exception {
+        List<Element<HearingBooking>> hearingDetails = createHearingBookings(date);
+
+        DynamicList dynamicHearingDates = draftCMOService.buildDynamicListFromHearingDetails(hearingDetails);
+
+        dynamicHearingDates
+            .setValue(
+                DynamicListElement.builder()
+                    .code(date.plusDays(5).toString())
+                    .label(date.plusDays(5).toString())
+                    .build());
+
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .data(ImmutableMap.of("cmoHearingDateList", dynamicHearingDates))
+                .build())
+            .build();
+
+        MvcResult response = makeRequest(request, "about-to-submit");
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = mapper.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+        assertThat(caseData.getCaseManagementOrder().getHearingDate())
+            .isEqualTo(date.plusDays(5).toString());
+    }
+
+    private List<String> getReturnedDatesFromResponse(AboutToStartOrSubmitCallbackResponse callbackResponse) {
```

Comment:
I don't think we need to know the methods params in the methods name. Perhaps just `getHearingDates`?

Reply:
changed as suggested